COMP: Simplify directory listing using vtksys::Directory

jcfr opened the pull request.

jcfr committed:
COMP: Simplify directory listing using vtksys::Directory

Connor-Bowley commented on RigidAlignment.cxx:
```diff
@@ -111,24 +108,29 @@ int main(int argc, char* argv[])
-      }
+      continue;
+    }
+    if(filename.find(suffix) != string::npos && filename.find_last_of(suffix) == filename.size() - 1)
```
I don't think the logic here is actually correct (the original logic, which your change replicates).

`std::string::find_last_of` has the following semantics (from https://en.cppreference.com/w/cpp/string/basic_string/find_last_of)

> Finds the last character equal to **_one of_** characters in the given character sequence.

The current use of this is `if(! getListFile(landmark.c_str(), landmarkList, "fcsv"))` which makes it seem like we are looking for a `.fcsv` file, but in reality a `.mov` file would incorrectly match.

I think this needs to change to use something like `std::string::rfind` instead.

Connor-Bowley replied:
I have updated this to be correct (although it doesn't use rfind, as that is more inefficient than necessary).

Connor-Bowley commented on RigidAlignment.cxx:
```diff
@@ -111,24 +108,29 @@ int main(int argc, char* argv[])
-    cerr<<"The directory does not exist! "<<path<<endl;
-    return false;
   }
+  sort(list.begin(), list.begin() + list.size());
```
If we are changing things anyway can we switch this to be `sort(list.begin(), list.end())` to be more idiomatic and clear we are sorting the whole list.

jcfr committed:
STYLE: Simplify sorting logic in getListFile()

Co-authored-by: Connor Bowley <connor.bowley@kitware.com>

Commit:
BUG: Fix string ends with logic

Existing implementation of endswith was incorrect due to use of
find_last_of, which operates on individual characters, not strings.

jcfr commented:
Thanks for helping out with this 🙏
